nucypher/policy/models.py

```diff
@@ -216,11 +210,17 @@ def __assign_kfrags(self) -> Generator[Arrangement, None, None]:
 
         for kfrag in self.kfrags:
             for arrangement in self._accepted_arrangements:
-                if kfrag not in self._enacted_arrangements:
+                if not arrangement in self._enacted_arrangements.values():
                     arrangement.kfrag = kfrag
-
                     self._enacted_arrangements[kfrag] = arrangement
+                    print(kfrag, arrangement, arrangement.ursula)
```

Comment:
Why `print` though? Is it a debug message?

Reply:
Ah ha... yeah that's not supposed to be there.